src/components/ThreadsList/ThreadsList.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import {Box, Divider, Button, Chip, Grid, Typography} from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import ThreadInfo from "./ThreadInfo";
+
+// Just for testing purpose
+type dummyThread = {
+    title: string,
+    author: string,
+    lastReply: string,
+    replyTime: string,
+    creationTime: string
+}
+
+type ThreadsListProps = {
+    threads: Array<dummyThread>
+}
+
+const ThreadsList : React.FC<ThreadsListProps> = (props) => {
+    const {
+        threads
+    } = props;
+
+    const firstColumnSize = 5;
+    const headers = [
+        "Author",
+        "Last replied by",
+        "Reply time",
+        "Creation time",
+        "Options"
+    ];
+
+    const headerElement = (sm: any, variant: any, color: any, text: string) => {
```

Comment:
Do not use `any` it is an antipattern. If you don't know the type then use `unknown` :D I think that in this case we know all the types:
```
(sm: number, variant: string, color: string, text: string)
```

Maybe you can also add a flag to our `tsconfig` -> `noImplicitAny: true`

Reply:
Number can't be set to false, whereas "sm" attribute in grid can have false, number and string

Reply:
I didn't see the second call. You don't call this function with grid attribute, but:
1. first time with `firstColumnSize` which is a number
2. second time with `false`

If you plan to call it with `sm` value from grid then the type in function can be `sm: string | number | boolean`

Reply:
`boolean | GridSize | undefined` is correct one, will change to that

Reply:
So you plan to pass the sm from grid here?

Reply:
It will be left as is currently, I will only change "any" to more accurate types

Reply:
So then GridSize is not correct - you pass either number or false?

Reply:
> 
> 
> So then GridSize is not correct - you pass either number or false?

`boolean | GridSize | undefined` is copied directly from MaterialUI source code. If I delete GridSize there will be error

Reply:
Oo, ok. I was wrong, sorry :D